runtime/suggestion-storage.js

```diff
@@ -49,8 +49,17 @@ export class SuggestionStorage {
   _initStore(id, storageKey, callback) {
     let schema = new Schema({names: ['Suggestions'], fields: {current: 'Object'}});
     let type = Type.newEntity(schema);
-    const promise = this._arc._storageProviderFactory._storageForKey(storageKey)._join(
+    let storage = this._arc._storageProviderFactory._storageForKey(storageKey);
+    let promise;
+    // TODO: the _join method exists for firebase storage and either connects or constructs
+    // depending on whether the storage exists. For pouch, the connect method does it.
+    // Should have a unified interface for this.
+    if (storage._join) {
+      promise = storage._join(
         id, type, storageKey, /* shoudExist= */ 'unknown', /* referenceMode= */ false);
+    } else {
+      promise = storage.connect(id, type, storageKey);
```

Comment:
I think you should use construct for pouch, not connect?

Reply:
I think so too.  construct uses mustExist == false

```
    return this._join(id, type, keyFragment, false, referenceMode);
```

and connect uses mustExist == true:

```
    return this._join(id, type, key, true);
```

'unknown' doesn't exactly map, but it's more like baseStorageFor usage:

```
    const storagePromise = this._join(type.toString(), type.collectionOf(), key, 'unknown') as Promise<FirebaseCollection>;
```

which hits this code:

```
    if (shouldExist !== 'unknown' && shouldExist !== currentSnapshot.exists()) {
      return null;
    }
```

so yes, construct feels more apropos

Reply:
Perhaps call  construct and then

	  provider.referenceMode = false;

Reply:
addressed comment (the code is kinda ugly at the moment, so the storage api update will be a blessing). PTAL